test/test_broker.hpp

```diff
@@ -1585,20 +1642,34 @@ class test_broker {
                 retains_.erase(topic);
             }
             else {
+                std::shared_ptr<as::steady_timer> tim_message_expiry;
+                if(message_expiry_interval) {
+                    tim_message_expiry = std::make_shared<as::steady_timer>(ioc_, message_expiry_interval.value());
+                    tim_message_expiry->async_wait(
+                      [&, topic = MQTT_NS::force_move(topic)]
```

Comment:
```suggestion
                      [this, topic, wp = std::weak_ptr<as::steady_timer>(tim_message_expiry)]
```

topic is used after this point.